[ci] raydepsets: adding var substitution for depset config

elliot-barn opened the pull request:
adding env substitution capabilities to raydepsets to parameterize python version, CUDA version, and platform in the future

elliot-barn committed:
adding compile and unit tests

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
updating config

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
adding config for raydepsets

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
updating unit tests

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
updating tests to use temp dir

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
adding test setup

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
updating unit tests

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
adding test requirement and constraint files

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
formatting and updating unit tests

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
updating pytest flag

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
adding uv binary to path

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
copying all files to tmp dir

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
moving uv binary into cli

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
adding unit tests

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
updating unit test

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
adding unit test to update version and compile

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
including env substitution for depset config

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
changing return type

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
updating unit tests

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
refactoring env substitution

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
adding tests for env substitution

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

aslonnie commented:
some other PR's are merged. please resolve the conflicts?

elliot-barn committed:
refactoring config args

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
refactoring var substitution

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
removing log

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
updating unit tests

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

aslonnie commented on ci/raydepsets/BUILD.bazel:
```diff
@@ -32,7 +34,7 @@ py_test(
         "test_data/requirements_compiled_test_update.txt",
         "test_data/requirements_test.txt",
-        "test_data/test.config.yaml",
+        "test_data/test.depsets.yaml",
```
could split the `*.config.yaml` -> `*.depsets.yaml` renaming into another separate PR?

elliot-barn replied:
Renamed here: https://github.com/ray-project/ray/pull/55198

aslonnie reviewed (commented):
I read the PR (roughly); let's talk on monday when you are back.

elliot-barn committed:
renaming config args

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
renaming config

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
more renaming

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

aslonnie committed:
[wheel] do not depend on `:ray_pkg` from java (#55181)

depends on `//:ray_pkg_zip` instead, with comment asking user to
manually run `//:gen_ray_pkg`

the java test rules are not conventional, hermetic ones, but "local"
ones that run in the source code directory. the ray core rules are all
converted to non-local, hermetic build rules, and we are deprecating the
local `ray_pkg` one. as a result, for the java rules to run, one has to
manually run the `bazel run //:gen_ray_pkg` to copy the ray core bits
back to the source directory.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>

eicherseiji committed:
[llm] bump vLLM to version 0.10.0 (#55067)

- Use upstream `RayPrometheusStatLogger` to close spec. decode + lora
errors
- Include fix for https://github.com/vllm-project/vllm/issues/20647
  - Restore PP=2 to DeepSeek-V2-Lite release test
  - Remove copy of `FrontendArgs` upstreamed with https://github.com/vllm-project/vllm/pull/20206/

Closes https://github.com/ray-project/ray/issues/54952

Includes fix for https://github.com/ray-project/ray/issues/54812

---------

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

aslonnie reviewed (commented):
could you resolve the merge conflicts?

ryanaoleary committed:
[Core] Remove `RAY_LOG(ERROR` from LabelSelector and validate in `prepare_label_selector` (#52964)

This PR is a follow-up to this comment:
https://github.com/ray-project/ray/pull/51901#discussion_r2070835498.
This PR changes the cluster resource scheduler to propagate a Ray status
to `ComputeResources` in `TaskSpecification` when the LabelSelector data
type is initialized. This allows a task built with a malformed label
selector to return an error as a more useful Python exception rather
than crashing Ray components in the C++.

https://github.com/ray-project/ray/issues/51564

---------

Signed-off-by: Ryan O'Leary <ryanaoleary@google.com>

dayshah committed:
[core] Revamp GetAllNodeInfo to be more efficient and take multiple NodeID's (#55115)

Needed for https://github.com/ray-project/ray/pull/55112 as
GetAllNodeInfo needs to be able to take multiple node id's as arguments
so you can get information for multiple nodes in one rpc to the GCS.
Revamping the request here to make it make more sense with this model
where you can select multiple nodes or get all nodes if you don't use
any selectors. Also adding an "optimized path" where if all the
selectors are just node id's, we can just do lookups in the map and add
them to the reply instead of iterating through all of it. Also adding
some extra tests for new behavior.

---------

Signed-off-by: dayshah <dhyey2019@gmail.com>

aslonnie reviewed (commented):
seems that rebasing failed..
